backend/prompt_studio/prompt_studio_registry/prompt_studio_registry_helper.py

```diff
@@ -190,7 +189,7 @@ def update_or_create_psr_tool(
                 logger.info(f"PSR {obj.prompt_registry_id} was created")
             else:
                 logger.info(f"PSR {obj.prompt_registry_id} was updated")
-
+            obj.modified_by = custom_tool.modified_by
```

Comment:
@johnyrahul how is this different from setting it in the `update_or_create()` method? I didn't get full context of the issue, however @harini-venkataraman please review if a similar pattern is observed anywhere else to fix that as well

Reply:
How the Django framework determines whether it should be updated or created is based on the fields supplied.
So when the shared user exports the tool  `modfied_by` file will be different from the owner. Hence it decides to create a new record instead of an update, which will generate an integrity issue
 @chandrasekharan-zipstack  makes sense?

Reply:
@harini-venkataraman then I suggest to look out for `update_or_create()` and `get_or_create()`

Reply:
@chandrasekharan-zipstack We mainly use it in account related logic and Output manager. Since this involves users, Integrity error is raised. Anyway will have to look out for other references.